macOS/LocalPackages/DataBrokerProtection/Sources/DataBrokerProtection/Shared/Operations/DataBrokerProfileQueryOperationManager.swift

```diff
@@ -113,127 +114,85 @@ struct DataBrokerProfileQueryOperationManager: OperationsManager {
                                    shouldRunNextStep: @escaping () -> Bool) async throws {
         Logger.dataBrokerProtection.log("Running scan operation: \(brokerProfileQueryData.dataBroker.name, privacy: .public)")
 
-        guard let brokerId = brokerProfileQueryData.dataBroker.id, let profileQueryId = brokerProfileQueryData.profileQuery.id else {
+        // 1. Validate that the broker and profile query data objects each have an ID:
```

Comment:
I'm curious to know when this wouldn't be the case. I'd potentially be up for removing this depending on the answer (although would likely require further changes).
I guess what unsettles me is, how did we get to this stage with that being the case? 

It's possibly that we never do. I'm perhaps overly alert to it because the codebase has a lot of other instances of "defensive" programming, some that were a much bigger deal, so it's perhaps worth validating. 

At any rate, if this should happen, I think we should handle it differently. If it shouldn't happen, I think we should handle it differently (e.g. at least an assert)

Reply:
Yeah this is something I want to change as part of the larger refactoring, I don't know that this function necessarily needs to care about this particular case, it should really be an up front prerequisite to _all_ jobs and considered a high priority issue if we ever detect it in a pixel, based on my understanding that these should always have IDs.